Makefile

```diff
@@ -1,115 +1,98 @@
-.PHONY: uninstall install setup-cli get-master-ip launch-dcos detroy-dcos docker-build docker kubectl-tunnel deploy
+.PHONY: azure aws gce uninstall install get-cli setup-cli get-master-ip get-master-elb_ip plan-dcos launch-dcos detroy-dcos kubectl-config kubectl-tunnel plan deploy
 
 RM := rm -f
-LAUNCH_CONFIG_FILE := launch.yaml
-CLUSTER_INFO_FILE := cluster_info.json
-BUILD_FILE := .output
-MASTER_IP_FILE := .master_ip
-ID_FILE := .id_key
 SSH_USER := core
-DCOS_INSTALLER_URL := https://downloads.dcos.io/dcos/stable/1.10.1/dcos_generate_config.sh
-
-include ./resources/gce.template.mk
-include ./resources/aws.template.mk
-include ./resources/azure.template.mk
+MASTER_IP_FILE := .master_ip
+MASTER_ELB_IP_FILE := .master_elb_ip
+TERRAFORM_INSTALLER_URL := github.com/dcos/terraform-dcos
+DCOS_LAUNCH_VERSION := 0.5.7
+KUBERNETES_VERSION := v1.7.10
+
+get-cli:
+	$(eval export DCOS_LAUNCH_VERSION)
+	$(eval export KUBERNETES_VERSION)
+	scripts/get_cli
+
+azure: clean
+	mkdir .deploy
+	cd .deploy; \
+	cp ../resources/desired_cluster_profile.azure desired_cluster_profile; \
+	terraform init -from-module $(TERRAFORM_INSTALLER_URL)//azure
+
+aws: clean
+	mkdir .deploy
+	cd .deploy; \
+	cp ../resources/desired_cluster_profile.aws desired_cluster_profile; \
+	terraform init -from-module $(TERRAFORM_INSTALLER_URL)//aws
+
+gce: clean
+	$(RM) -r .deploy
+	mkdir .deploy
+	cd .deploy; \
+	cp ../resources/desired_cluster_profile.gce desired_cluster_profile; \
+	terraform init -from-module $(TERRAFORM_INSTALLER_URL)//gcp; \
+	rm desired_cluster_profile.tfvars.example
 
 install:
 	dcos package install --yes beta-kubernetes
 
-
 uninstall:
 	dcos package uninstall --yes beta-kubernetes
 
-
-DCOS_USERNAME := bootstrapuser
-DCOS_PASSWORD := deleteme
-setup-cli: kubectl-config
+setup-cli:
 	$(call get_master_ip)
 	dcos cluster setup http://$(MASTER_IP)
 
-kubectl-config:
-	kubectl config set-cluster dcos-k8s --server=http://localhost:9000
-	kubectl config set-context dcos-k8s --cluster=dcos-k8s --namespace=default
-	kubectl config use-context dcos-k8s
-
 get-master-ip:
 	$(call get_master_ip)
 	@echo $(MASTER_IP)
 
 define get_master_ip
 $(shell test -f $(MASTER_IP_FILE) || \
-	dcos-launch describe | jq '.["masters"] | .[] | .["public_ip"]' | head -1 | sed s/\"//g | sed s/\:[0-9]*//g > $(MASTER_IP_FILE))
+	terraform output -state=.deploy/terraform.tfstate "Mesos Master Public IP" | head -1 > $(MASTER_IP_FILE))
 $(eval MASTER_IP := $(shell cat $(MASTER_IP_FILE)))
 endef
 
-launch-dcos: $(CLUSTER_INFO_FILE)
-	dcos-launch wait
+get-master-elb-ip:
```

Comment:
ELB is unique to AWS. Does this only work for AWS? What about Azure and GCP?

Reply:
it is just the naming in TF, it works in azure and gcp

Reply:
OK so please rename the goal

Reply:
addressed